Userland/Libraries/LibJS/JIT/Compiler.cpp

```diff
@@ -1239,6 +1239,64 @@ void Compiler::compile_get_by_id(Bytecode::Op::GetById const& op)
     branch_if_object(ARG1, [&] {
         extract_object_pointer(GPR0, ARG1);
 
+        // NOTE: Fast path for Array.length which magically reflects
+        //       the "array-like size" of the array object's property storage.
+        Assembler::Label no_magical_length_property_case;
+        if (m_bytecode_executable.get_identifier(op.property()) == "length"sv) {
+            // if (!object.has_magical_length_property) goto no_magical_length_property_case;
+            m_assembler.mov8(
+                Assembler::Operand::Register(GPR1),
+                Assembler::Operand::Mem64BaseAndOffset(GPR0, Object::has_magical_length_property_offset()));
+            m_assembler.jump_if(
+                Assembler::Operand::Register(GPR1),
+                Assembler::Condition::EqualTo,
+                Assembler::Operand::Imm(0),
+                no_magical_length_property_case);
+
+            // NOTE: The base object has a magical "length" property, so now we just need
+            //       to extract the "array-like size" from the object property storage.
+            //       If we run into any issues, we'll jump to the slow case and figure things out in C++.
+
+            // GPR0 = object->indexed_properties().storage()
+            m_assembler.mov(
+                Assembler::Operand::Register(GPR0),
+                Assembler::Operand::Mem64BaseAndOffset(GPR0, Object::indexed_properties_offset() + IndexedProperties::storage_offset()));
```

Comment:
I may be wrong, but this trashes GPR0, right?
But there are two checks below that jump to the slow case, and there we have
https://github.com/awesomekling/serenity/blob/fd3e611ab222e837c61473fd2dfb4bd76dfe27f6/Userland/Libraries/LibJS/JIT/Compiler.cpp#L1301-L1303
which reads GPR0, no?

Reply:
The slow case is all the way down here: https://github.com/awesomekling/serenity/blob/fd3e611ab222e837c61473fd2dfb4bd76dfe27f6/Userland/Libraries/LibJS/JIT/Compiler.cpp#L1387

Reply:
Ah, my bad, thanks for clarifying